e2e/main_test.go

```
@@ -0,0 +1,134 @@
+	}
+	// create the clientset
+	client, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	if err := ready(client, *expectedNodes); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// createNamespace
+	if _, err := createNamespace(client, namespace); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	// run tests
+	exitCode := m.Run()
+
+	if err := deleteNamespace(client, namespace); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	os.Exit(exitCode)
+}
+
+func createNamespace(c kubernetes.Interface, name string) (*v1.Namespace, error) {
+	namespace, err := c.CoreV1().Namespaces().Create(&v1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create namespace with name %v %v", name, namespace)
+	}
+	return namespace, nil
+}
+
+func deleteNamespace(c kubernetes.Interface, name string) error {
+	return c.CoreV1().Namespaces().Delete(name, nil)
+}
+
+// Ready blocks until a the cluster is considered available. The current
+// implementation checks that the expected number of nodes are registered.
+func ready(c kubernetes.Interface, expectedNodes int) error {
+	f := func() error {
+		list, err := c.CoreV1().Nodes().List(metav1.ListOptions{})
+		if err != nil {
+			return err
+		}
+
+		if len(list.Items) != expectedNodes {
+			return fmt.Errorf("cluster is not ready, expected %v nodes got %v", expectedNodes, len(list.Items))
+		}
+
+		for _, node := range list.Items {
+			for _, condition := range node.Status.Conditions {
+				if condition.Type == v1.NodeReady {
+					if condition.Status != v1.ConditionTrue {
+						return fmt.Errorf("One or more nodes not in the ready state: %v", node.Status.Phase)
+					} else {
+						break
+					}
+				}
+			}
+		}
+
+		return nil
+	}
+
+	if err := retry(12, 10*time.Second, f); err != nil {
+		return err
+	}
+	return nil
+}
+
+func retry(attempts int, delay time.Duration, f func() error) error {
```

It's the same thing here right? The `err` returned by `f()` is ignored.

In `Poll()` you can achieve the same by ignoring the errors, then the only error can happen will be the timout.